### PR TITLE
Major changes for two-scale analysis

### PR DESCRIPTION
**Pull request by dmcglinn**
Here is a summary of the key changes: 
* Metric "pct_rare" was added which records how many species in a community have fewer than 5% of the individuals in the community. This replaces the default metric of rarity which was previously S_asymp. This change was performed because S_asymp his highly correlated with S (McGill 2011). 
* Group scale multi category treatment test now implements test as described in #171 
* The function `rarefaction` now extrapolates S using the chao1 approaches and this is used when computing rarefied S (i.e., `S_n`)
* Rarefied S is no longer `S_rare` it is now `S_n`. 
* The package no longer has a dependency on the package `vegan`

### CODE REVIEW

File: R/mobr.R [hunk 1]
```diff
@@ -209,17 +212,31 @@ sphere_dist = function(long, lat){
...
 #' @return A vector of rarefied species richness values
 #' @author Dan McGlinn and Xiao Xiao
-#' @references Cayuela, L., N.J. Gotelli, & R.K. Colwell (2015) Ecological and 
-#' biogeographic null hypotheses for comparing rarefaction curves. Ecological
-#' Monographs 85:437-454. Appendix A: 
-#' http://esapubs.org/archive/mono/M085/017/appendix-A.php
-#' 
+#' @references 
+#' Cayuela, L., N.J. Gotelli, & R.K. Colwell (2015) Ecological and 
+#'  biogeographic null hypotheses for com.paring rarefaction curves. Ecological
```
Comment (FelixMay) [inline R/mobr.R]: Small type in line 229 "comparing"

File: DESCRIPTION [hunk 2]
```diff
@@ -15,7 +15,6 @@ Depends:
     scales,
     dplyr
 Imports:
-    vegan,
```
Comment (FelixMay) [inline DESCRIPTION]: In `mobsim` I neve had problems with the `vegan` dependency. But I think in general the less dependencies, the better

File: R/mobr_boxplots.R [hunk 3]
```diff
@@ -74,570 +93,590 @@ calc_PIE = function(x) {
...
+   
+    abund_group = aggregate(abund_mat, by = list(group_id), FUN = "sum")
+   
+    dat_groups = calc_biodiv(abund_mat = abund_group[ , -1],
+                             groups = abund_group[ , 1],
+                             index = index, effort = effort, 
+                             rare_thres = rare_thres)
+    delta_groups = dat_groups %>%
+                   group_by(index, effort) %>%
+                   summarise(delta = mean(dist(value))) %>%
```
Comment (FelixMay) [inline R/mobr_boxplots.R]: Smart implementation of all pairwise differences!

**Comment by dmcglinn** [comment]
Thanks for fast feedback @FelixMay, I will make the requested changes and update the PR. Also thank you for not minding the switch from `<-`  to `=` - I know this is a bit silly. Most R folks prefer `<-` over `=` but I just want the code style to be internally consistent in the package and I prefer `=`.  I also just wanted to note that this PR addresses comments raised in #170 regarding the `D0.hat` function and interpolation of rarefaction curves.

File: DESCRIPTION (hunk 2 above)
Comment (dmcglinn) [reply DESCRIPTION]: yea I love vegan but it makes our package more likely to break in the future as 3rd party code changes.

File: R/mobr.R (hunk 1 above)
Comment (dmcglinn) [reply R/mobr.R]: got this in the spelling check thanks

**Pull request merged**